src/structures/Message.js

```diff
@@ -155,6 +148,10 @@ class Message {
      * @private
      */
     this._edits = [];
+
+    if (data.member && this.guild && this.author) {
+      this.guild.members.set(data.member.id, Object.assign(data.member, { user: this.author }));
```

Comment:
This would overwrite existing members breaking their references.
This also would set a plain object and not a `GuildMember` instance.

As I do not think that Discord propagates changes through this, a simple check whether the member is uncached and then using `this.guild._addMember(..., false)` should do the trick.

Reply:
Resolved by https://github.com/discordjs/discord.js/pull/3684/commits/59a47a7bf52503909ba8b70e92f632a4ab1e9e72 (I hope)